src/components/text-editor/__internal__/utils/utils.spec.ts

```diff
@@ -70,14 +70,17 @@ describe("getSelectedLength", () => {
       text: "foo",
       characterList: List(),
     });
-
-    const newBlockMap = content.getBlockMap().set(newBlock.key, newBlock);
+    const newBlockMap = content
+      .getBlockMap()
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .set((newBlock as any).key, newBlock);
```

Comment:
```suggestion
      .set(newBlock.getKey(), newBlock);
```

I know nothing about DraftJS and its types, but disliking the cast to `any` here and looking up the docs, I saw this alternative and it seems to work for me without TS complaining (and the test still passes). I don't knowwhy accessing `.key` directly doesn't work (but I guess it's private and DraftJS wants to force you to use the `getKey` method), but anyway I prefer this to using `any`.

Reply:
this hasn't yet been addressed btw